include/deal.II/matrix_free/task_info.h

```diff
@@ -0,0 +1,518 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2011 - 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+#ifndef __deal2__matrix_free_task_info_h
+#define __deal2__matrix_free_task_info_h
+
+
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/memory_consumption.h>
+#include <deal.II/base/index_set.h>
+#include <deal.II/base/utilities.h>
+#include <deal.II/base/tensor.h>
+#include <deal.II/base/thread_management.h>
+#include <deal.II/base/vectorization.h>
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+
+
+namespace internal
+{
+  /**
+   * An interface for the worker object that runs the various operations we
+   * want to perform during the matrix-free loop.
+   *
+   * @author Katharina Kormann, Martin Kronbichler, 2018
+   */
+  struct MFWorkerInterface
+  {
+  public:
+    virtual ~MFWorkerInterface() {}
```

Comment:
New internal interface has been introduced to avoid multiple function pointers in DG case (for cell, inner and boundary faces).